server/src/main/java/org/candlepin/async/tasks/HealEntireOrgJob.java

```
@@ -0,0 +1,176 @@
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.common.filter.LoggingFilter;
+import org.candlepin.controller.AutobindDisabledForOwnerException;
+import org.candlepin.controller.Entitler;
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+import org.candlepin.resource.dto.AutobindData;
+
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+import org.jboss.resteasy.spi.BadRequestException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xnap.commons.i18n.I18n;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * HealEntireOrgJob
+ */
+public class HealEntireOrgJob implements AsyncJob {
+
+    private static Logger log = LoggerFactory.getLogger(HealEntireOrgJob.class);
+    public static final String JOB_KEY = "HEAL_ENTIRE_ORG_JOB";
+    private static final String JOB_NAME = "heal entire org job";
+    public static final String OWNER_KEY = "org";
+    public static final String ENTITLE_DATE_KEY = "entitle_date";
+
+    private OwnerCurator ownerCurator;
+    private Entitler entitler;
+    private ConsumerCurator consumerCurator;
+    private I18n i18n;
+
+    @Inject
+    public HealEntireOrgJob(Entitler entitler, ConsumerCurator consumerCurator, OwnerCurator ownerCurator,
+        I18n i18n) {
+        this.entitler = entitler;
+        this.consumerCurator = consumerCurator;
+        this.ownerCurator = ownerCurator;
+        this.i18n = i18n;
+    }
+
+    @Override
+    public Object execute(JobExecutionContext context) throws JobExecutionException {
+        try {
+
+            final JobArguments arguments = context.getJobArguments();
+            final String ownerKey = arguments.getAsString(OWNER_KEY);
+            Owner owner = ownerCurator.getByKey(ownerKey);
```

We should probably be checking that we actually received an owner here, and failing more gracefully (and terminally)

---

If the owner does not exist then should we throw an exception or just simply terminate the method execution?

---

In this case, an exception is probably the way to go. I would assume that the process that fired off the job already validated the org existed at the time it was queued, so this would be a pretty nasty race condition if it happened.

Safest way to handle this is likely to throw a terminal JobException.